torch/_meta_registrations.py

```diff
@@ -5192,6 +5193,154 @@ def meta__scaled_dot_product_flash_backward(
     return grad_q, grad_k, grad_v
 
 
+@register_meta(
+    [
+        aten._scaled_dot_product_flash_attention_mask,
+    ]
+)
+def meta__scaled_dot_product_flash_mask(
+    query: Tensor,
+    key: Tensor,
+    value: Tensor,
+    dropout_p: float = 0.0,
+    is_causal: bool = False,
+    return_debug_mask: bool = False,
+    attn_mask: Optional[Tensor] = None,
+    scale: Optional[float] = None,
+):
```

Comment:
What's the difference from the version without mask: `meta__scaled_dot_product_flash`? Can we reuse that logic without re-implementing?

Reply:
The only difference is that the masked version has an optional variable `attn_mask`. Re-implemented by calling `meta__scaled_dot_product_flash_mask` in `meta__scaled_dot_product_flash`.